ivy/functional/frontends/builtins/math/power_and_logarithmic_functions.py

```diff
@@ -0,0 +1,31 @@
+import ivy
+from ivy.functional.frontends.builtins.func_wrapper import (
+    from_zero_dim_arrays_to_scalar,
+)
+
+
+@from_zero_dim_arrays_to_scalar
+def sqrt(x):
+    return ivy.sqrt(x)
+
+
+@from_zero_dim_arrays_to_scalar
+def pow(x, y):
+    return ivy.pow(x, y)
+
+
+@from_zero_dim_arrays_to_scalar
+def log(x, base=None):
+    if base:
+        return ivy.divide(ivy.log(x), ivy.log(base))
+    return ivy.log(x)
+
+
+@from_zero_dim_arrays_to_scalar
+def log2(x):
+    return ivy.log2(x)
+
+
+@from_zero_dim_arrays_to_scalar
+def exp(x):
+    return ivy.exp(x)
```

Comment:
Likewise shouldn't we add `@to_ivy_arrays_and_back` in these functions as well?

Reply:
This function from `math` built-in module also accepts scalars and returns scalars